### PR TITLE
Testing

### PR DESCRIPTION
**Pull request by Howon**

### CODE REVIEW

File: test/oscar/compiler/add.oscar [hunk 1]
```diff
@@ -0,0 +1,3 @@
+def main() => unit = {
+  println(5 + 5);
+}
```
Comment (jibben) [inline test/oscar/compiler/add.oscar:3]: newline?

File: test/oscar/compiler/arithFunctionChaining.oscar [hunk 2]
```diff
@@ -0,0 +1,15 @@
+def f1(num: int) => int = {
+  return int * 2;
```
Comment (jibben) [inline test/oscar/compiler/arithFunctionChaining.oscar]: return num \* 2

File: test/oscar/compiler/arithFunctionChaining.oscar [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+def f2(num: int) => int = {
+  return int + 5;
```
Comment (jibben) [inline test/oscar/compiler/arithFunctionChaining.oscar]: return num + 5

File: test/oscar/compiler/arithFunctionChaining.oscar [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+def f3(num: int) => int = {
+  return int - 20;
```
Comment (jibben) [inline test/oscar/compiler/arithFunctionChaining.oscar]: return num - 20

File: test/oscar/scanner/types.oscar [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  mut pool<Worker> mWorkerPool;
+
+  def println(s: string) => unit = { }
+
+  def apply(f: (double) => double, a: double) => double = {
+    return f(a);
+  }
+}
+
+def main() = unit => {
```
Comment (jibben) [inline test/oscar/scanner/types.oscar]: def main() => unit =

File: test/oscar/scanner/types.oscar [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  mut int mI;
+  mut double mD;
+  mut boolean mB;
+  mut char mC;
+  mut string mS;
+  mut list<int> mIntList;
+  mut set<int> mIntSet;
+  mut map<int, double> mIntDoubleMap;
+  mut tuple<string, int, double> mStrIntDoubleTuple;
+  mut pool<Worker> mWorkerPool;
```
Comment (jibbenHillen) [inline test/oscar/scanner/types.oscar]: Is this something we want to allow? Does it make sense to have a mutable pool? What would this even entail

File: Makefile [hunk 7]
```diff
@@ -0,0 +1,64 @@
+TARGET=oscar
```
Comment (jibbenHillen) [inline Makefile]: This MAKEFILE is tested and works correctly, right?

Also might be worth adding an `all` target

**Review by jibben — DISMISSED** [review]

File: test/oscar/compiler/arithFunctionChaining.oscar (hunk 2 above)
Comment (jibben) [reply test/oscar/compiler/arithFunctionChaining.oscar]: Actually none of the test files end in newline. For our language's style guideline, do we want to have a trailing newline or no?

File: test/oscar/compiler/add.oscar (hunk 1 above)
Comment (Howon) [reply test/oscar/compiler/add.oscar:3]: thx

File: test/oscar/compiler/arithFunctionChaining.oscar (hunk 2 above)
Comment (Howon) [reply test/oscar/compiler/arithFunctionChaining.oscar]: idk should we?

File: test/oscar/compiler/arithFunctionChaining.oscar (hunk 3 above)
Comment (Howon) [reply test/oscar/compiler/arithFunctionChaining.oscar]: im too deep entrenched in java world

**Comment by Howon** [comment]
@jibbenHillen  @xvladus1 @anthonyholley @aethos someone approve this

**Review by aethos — DISMISSED** [review]

File: test/test.sh [hunk 8]
```diff
@@ -147,14 +149,14 @@ make_oscar
...
-scanner_extension=.scan
+#test_path="$TEST_DIR"/oscar/scanner/
+#scanner_extension=.scan
 
 # test_scanner $test_path $scanner_extension
 
 test_path=oscar/compiler/
-compiler_extension=.out
 test_extension=.oscar
+compiler_extension=$test_extension.out
```
Comment (aethos) [inline test/test.sh:159]: this is breaking

File: test/oscar/compiler/helloWorld.oscar.out [hunk 9]
```diff
@@ -0,0 +1 @@
+Hello World
```
Comment (vlad-scherbich) [inline test/oscar/compiler/helloWorld.oscar.out:1]: does `println` print a newline after the string? is this something to check here?

**Review by aethos — APPROVED** [review]

**Comment by aethos** [comment]
i approve

File: test/oscar/compiler/helloWorld.oscar.out (hunk 9 above)
Comment (Howon) [reply test/oscar/compiler/helloWorld.oscar.out:1]: i thought so

File: Makefile (hunk 7 above)
Comment (Howon) [inline Makefile]: yes

File: test/oscar/scanner/types.oscar (hunk 5 above)
Comment (Howon) [reply test/oscar/scanner/types.oscar]: fixed

File: test/oscar/scanner/types.oscar (hunk 6 above)
Comment (Howon) [inline test/oscar/scanner/types.oscar]: add more actros? idk
im down to no mutable pools

**Comment by aethos** [comment]
@jibbenHillen

**Review by jibben — APPROVED** [review]

**Pull request merged**